build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -425,6 +425,19 @@ message Action {
   bool do_not_cache = 7;
 
   reserved 8; // Used for field moved to [Command][build.bazel.remote.execution.v2.Command].
+
+  // An optional additional salt value used to place this `Action` into a
+  // separate cache universe or "epoch" from other instances having the
+  // same field contents. This salt typically comes from operational
+  // configuration specific to sources such as repo and service configuration,
+  // and allows disowning an entire set of ActionResults that might have been
+  // poisoned by buggy software or tool failures.
+  //
+  // As an example for repo configuration, a repo could contain a special file
+  // such as "CacheEpoch.txt" containing an incrementing number, whose hash is
+  // added to this salt value. Repo owners could increment the number when
```

Comment:
Just curious, what is the benefit of adding the hash of an incrementing number, compared to adding the number itself?

Reply:
I agree that in the case of a simple rotating integer salt using the digest provides no additional value. It's nice to have flexibility beyond just using an integer, though. A more advanced system might, for example, try to capture the version of the kernel and installed packages, add a numeric salt, and use the digest of all of that here. (This would obviously make the cache namespace specific to machines with identical configuration, which may or may not be desirable.)

Reply:
We actually hash the full file contents as some teams leave comments why they changed things. Some teams also like a major.minor string instead of a counting integer. We leave it up to them, as long as a change in the file is unique across the lifetime of the cache entries (weeks, typically).